src/ok.rs

```diff
@@ -135,11 +136,15 @@ const SEV_ES_CPUIDS: &[CpuId; 1] = &[CpuId {
 pub fn cmd(gen: Option<SevGeneration>, quiet: bool) -> Result<()> {
     let mut passed = true;
 
+    // Collect all tests.
     let cpuid_vec = match gen {
         Some(g) => collect_cpuids(g),
         None => collect_cpuids(SevGeneration::Es),
     };
+    let sysfs_tests: Vec<Box<dyn Fn(&mut fs::OpenOptions) -> TestResult>> =
+        vec![Box::new(dev_sev_rw), Box::new(dev_sev_rw)];
```

Comment:
I would just split `dev_sev_rw` into `dev_sev_r` and `dev_sev_w` and you can still have a common core function, say `dev_sev_rw`. Then, just have the vec contain `dev_sev_r` and `dev_sev_w`.

`dev_sev_r` and `dev_sev_w` would basically just be wrapper functions for `dev_sev_rw` but supplying the relevant fs::Options and w/e String you want to pair with the result message.

As it is right now, you're trying to share this behavior for both which is leaking out into the test runner code. There's no need to special-case the runner code over this.

Reply:
I see what you mean, I could also group the sysfs tests with the KVM and memlock tests by doing this.